neoexchange/core/views.py

```diff
@@ -3884,17 +3937,21 @@ def get(self, request, *args, **kwargs):
 class LCPlot(LookUpBodyMixin, FormView):
 
     template_name = 'core/plot_lc.html'
+    form_class = AddPeriodForm
 
-    def get(self, request, *args, **kwargs):
-        best_period = self.body.get_physical_parameters('P', False)
+    def get(self, request, form=None, *args, **kwargs):
+        period_list = PhysicalParameters.objects.filter(body=self.body, parameter_type='P')
+        best_period = period_list.filter(preferred=True)
         if best_period:
-            period = best_period[0].get('value', None)
+            period = best_period[0].value
```

Comment:
Could this fail if someone creates a period but doesn't fill in a `value` ? Or is that forbidden by the form validation ?

Reply:
It would violate the form validation. If someone did it manually, they might be able to break this.

Reply:
I don't think this will break. (Famous last words I'm sure.)
`best_period` is a queryset. If it's empty, `period = None`, if it's not empty, there will be a 0th element, that will either have a value that will become period, or it won't and `period = None` .